### PR TITLE
feat(wallet): query incremental timeout

### PR DESCRIPTION
**Pull request by kehiy**
## Description

This PR creates a time-out for querying data from gRPC on the wallet. we set a timeout for 5 seconds where the default is 30 seconds. we can make sure a slow RPC won't slow down the proccess.

### CODE REVIEW

File: wallet/client.go [hunk 1]
```diff
@@ -118,11 +129,14 @@ func (c *grpcClient) sendTx(trx *tx.Tx) (tx.ID, error) {
...
 
+	ctx, cancel := context.WithTimeout(c.ctx, time.Second*5)
+	defer cancel()
+
 	data, err := trx.Bytes()
 	if err != nil {
 		return hash.UndefHash, err
 	}
-	res, err := c.transactionClient.BroadcastTransaction(c.ctx,
+	res, err := c.transactionClient.BroadcastTransaction(ctx,
```
Comment (Ja7ad) [inline wallet/client.go:139]: Important things is gRPC method `BroadcastTransaction` should pass context to lower layers to stop process.

When you call gRPC make a go routine in background to processing gRPC method then your context send signal cancel base on timeout, process don't canceled in background.

**Review by Ja7ad — CHANGES_REQUESTED** [review]
for create context not required to call `context.WithoutCancel`.

```go
ctx := context.WithoutCancel(context.Background())
```

**Comment by kehiy** [comment]
> for create context not required to call `context.WithoutCancel`.
> 
> ```go
> ctx := context.WithoutCancel(context.Background())
> ```

I'm calling WithTimeout, not WithCancel.

File: wallet/client.go (hunk 1 above)
Comment (kehiy) [reply wallet/client.go:139]: Which layer I should pass it to?

Comment (kehiy) [reply wallet/client.go:139]: What should we do to cancel it on background? also, why it won't cancel in background when the timeout reaches?

**Comment by Ja7ad** [comment]
> > for create context not required to call `context.WithoutCancel`.
> > ```go
> > ctx := context.WithoutCancel(context.Background())
> > ```
> 
> I'm calling WithTimeout, not WithCancel.

line 29

**Comment by kehiy** [comment]
> > > for create context not required to call `context.WithoutCancel`.
> > > ```go
> > > ctx := context.WithoutCancel(context.Background())
> > > ```
> > 
> > 
> > I'm calling WithTimeout, not WithCancel.
> 
> line 29

this line is not included in this PR, let me update it.

**Comment by kehiy** [comment]
@Ja7ad fixed.

File: wallet/client.go [hunk 2]
```diff
@@ -76,7 +78,10 @@ func (c *grpcClient) getBlockchainInfo() (*pactus.GetBlockchainInfoResponse, err
 		return nil, err
 	}
 
-	info, err := c.blockchainClient.GetBlockchainInfo(c.ctx,
+	ctx, cancel := context.WithTimeout(c.ctx, time.Second*5)
```
[the remark this answers is not in the document]
Comment (kehiy) [reply wallet/client.go:81]: It seems this way of handling the issue it not correct based on @Ja7ad review. once we find the correct way I'll cleanly handle the code. this PR needs change.

[the remark this answers is not in the document]
Comment (kehiy) [reply wallet/client.go]: @b00f @Ja7ad 
can you recheck this?
Is there any close method to call cancel there? or should I run a routine to call the cancel based on a condition?

**Comment by Ja7ad** [comment]
@kehiy This method is not correct to solve the problem and I will implement another method to solve the problem.

**Pull request closed without merging**